compiler/instruction.go

```diff
@@ -113,26 +113,58 @@ type switchCaseIRInstruction struct {
 	writerIndex int
 	// If there is no target field, or the target is not a union, the readerIndex is -1
 	readerIndex int
+	needsJump   bool
+	insts       []vm.Instruction
+}
+
+// Creates a new switch-case IR instruction for the given switch, writer and reader IDs.
+// A 'rejecting case' instruction generates a pair (Exit, Null) and a Jump past the
```

Comment:
I'm not sure I follow the "rejecting case" - why would we make `(exit, null)` a special case when nulls are already handled by a sequence of `enter, (set, null), exit`?

Reply:
Because `(exit, null)` is a special case signalling the vm evaluator to not set the outer field.

If an optional union is to be deserialized, the VM would "enter" the union itself, prior to read the field describing the union type. When entering, the struct is actually created in memory, so its pointer is not nil. Once the union type field is read, if the union type is null, the VM would need to undo the creation of the union struct. The VM outer caller would know the union is valid ("accepted") if the inner field exits with a non-null value. Otherwise, the created struct would be discarded (nil'ed).